### DISCUSSION TITLE
chromium_arg in seleniumbase

### DISCUSSION THREAD

**Discussion by joako233** [seleniumbase/SeleniumBase#3038] [Q&A] [answered]
Hi Michael, I wanted to ask if the arguments I've set in the Chromium ARG could cause any issues when using Selenium Base. Do you think they might interfere with its functionality?

chromium_arg = '--blink-settings=imagesEnabled=false,--enable-quic,--enable-http2,--disable-background-timer-throttling,--disable-backgrounding-occluded-windows,--disable-renderer-backgrounding'

and using this: with SB(is_mobile=True,headless2=True,browser='edge',chromium_arg=chromium_arg) as sb:

**Comment by mdmintz** [comment, marked as answer]
Looks like you're trying to set some args that have already been set by default, such as `disable-backgrounding-occluded-windows`: https://github.com/seleniumbase/SeleniumBase/blob/af3d9545473e55b2a25cdbab8be0b1ed5e1f6afa/seleniumbase/core/browser_launcher.py#L1844

Setting the same arg twice may cause issues.

Also, some args that you're trying to set will prevent UC Mode from working at all.

UC Mode is only for Chrome (and may work with some other Chromium browsers), but it won't work on Edge.
Mobile Mode will likely get you detected.
Headless mode will prevent `PyAutoGUI` from working, which is needed in some cases.

**Reply by joako233** [reply]
Okay Thanks!! Where on the SeleniumBase website can I see which arguments are added by default in SeleniumBase?

**Reply by mdmintz** [reply]
Look for them in [SeleniumBase/seleniumbase/core/browser_launcher.py](https://github.com/seleniumbase/SeleniumBase/blob/master/seleniumbase/core/browser_launcher.py)

**Reply by joako233** [reply]
Okay, and do you know how to solve the error: Timed out receiving message from renderer that I receive? Here is the discussion: https://github.com/seleniumbase/SeleniumBase/discussions/3040
Thanks

**Reply by mdmintz** [reply]
It depends on your situation, see various tickets: https://github.com/search?q=repo%3Aseleniumbase%2FSeleniumBase+Timed+out+receiving+message+from+renderer&type=issues

**Reply by joako233** [reply]
Hi, I've tried many things these last few days, among them:

>  pls="none", pls='eagle',"--dns-prefetch-disable"

. I've updated all the libraries, increased the timeout for the element I'm trying to find, but none of these worked. I'm still getting the same error: 

> "Timed out receiving message from renderer."

 I would appreciate it if you could take a look at it when you get a chance. Thanks in advance. This is the code:



```
with SB(is_mobile=True,headless2=True,chromium_arg=chromium_arg,pls="none") as sb:

    sb.driver.maximize_window()
    for url in urls:

        sb.execute_script(f"window.open('{url}');")
    
    time.sleep(1)

    ventanas = sb.driver.window_handles

    lista_buenas = []
    eventos = []
    #This is a loop to check on each tab that no captcha has appeared on the webpage I am visiting.
    while True:
    
        if len(lista_buenas) == len(urls):
        
            break
    
        for x in range(1,len(ventanas)):
        
            if x not in lista_buenas:
                sb.switch_to_window(sb.driver.window_handles[x])
                try:
                    sb.driver.switch_to_frame("iframe[sandbox='allow-scripts allow-same-origin allow-forms']")
                    sb.refresh()
                except:
                    nombrito = sb.get_text("//header[@data-pw-test-id='event-header']//h1/span")
                    print(f"(+) Todo bien en {nombrito}")
                    eventos.append(nombrito)
                    lista_buenas.append(x)
            else:
                continue

    sb.switch_to_window(sb.driver.window_handles[0])
    sb.driver.close()
    time.sleep(1)
    nuevas = sb.driver.window_handles

   
    #this is the main code

    while True:
    
        try:
            for x in range(len(nuevas)):
            
                sb.switch_to_window(nuevas[x])
                #sb.save_screenshot('hormiga.png')
        
                sb.click("//button[@data-button-tracking-id='toggle-tickets-modal-open']",scroll=False)

                try:
                    sb.switch_to_frame("//iframe[@id='#tickets-iframe-s']")
                    #sb.save_screenshot('probando.png')
            
                except:
                    print("(-) HA TARDADO EN CARGAR...")
                    pass    
            
                try:
                    #sb.assert_element_present("//div[@id='buynow']",timeout=1.5)
                    sb.assert_element_present("//li[@class='onsale but']//div[contains(text(),'32,80 €')]",timeout=1.5)
                    nombre_evento = eventos[x]
        
                    mensaje = f"Entrada encontrada para el evento: {nombre_evento}"
                    telegram(mensaje)
                    time.sleep(25)
                    sb.switch_to_default_content()
                    sb.click('//button[@data-button-tracking-id="toggle-tickets-modal-closed"]')
                except:
                    sb.switch_to_default_content()
                    sb.click('//button[@data-button-tracking-id="toggle-tickets-modal-closed"]')
                    continue
        except Exception as e:
            print(e)
            input()
            break
  

```
